Homework4/Task1/MatrixTracker.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+namespace Task1
+{
+    public class MatrixTracker<T>
+    {
+        public DiagonalMatrix<T> MatrixReceived { get; }
+
+        public MatrixTracker(DiagonalMatrix<T> diagonalMatrix)
+        {
+            MatrixReceived = diagonalMatrix;
+            //here we subscribe to the event
+            MatrixReceived.ElementChangedHandler += MatrixReceived.Anouncement;
```

Comment:
If you made event handler as a method of MatrixTracker instead of DiagonalMatrix,
then you can store UndoArgs in MatrixTracker and reuse them.

Reply:
Again, move handler from DiagonalMatrix to MatrixTracker and store args in MatrixTracker. 

  MatrixReceived.ElementChangedHandler += **MatrixTracker.OnElementChanged**;